.circleci/config.yml

```diff
@@ -30,13 +30,38 @@ jobs:
       - run: docker version
       - run: nproc
       - run: make clear-images
-      - run: make full-test -j $(nproc)
+
+      # Work out what needs rebuilding and what has changed
+      - run: |
+          ( ./changed-since-last.sh
+            echo "export DIND_CHANGED DIND_TAG DIND_PREV"
+            echo "export FN_CHANGED FN_TAG FN_PREV"
+          ) >> $BASH_ENV
+
+      # Rebuild the build tools if necessary
+      # This leaves local docker images around
+      - run: |
+          if [[ -n "$DIND_CHANGED" ]]; then
+            make build-dind
+          fi
+
+      # Rebuild fnserver if necessary
+      - run: |
+          if [[ -n "$FNLB_CHANGED" ]]; then
```

Comment:
think this is either misleading, wrong or cruft from removal :)

Reply:
`FN_CHANGED` right?

Reply:
actually I think we should probably do `make full-test` unconditionally really ?

Reply:
The changed-since-last.sh script will set the one on the basis of the other. I'll rename these to make that more obvious.